src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,48 @@
+package duke.task;
+
+public class Task {
+    protected String taskDescription;
+    protected boolean isDone;
+
+    public Task(String taskDescription) {
+        this.taskDescription = taskDescription;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void doTask() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks the task as completed.
+     */
+    public void undoTask() {
+        this.isDone = false;
+    }
+
+    /**
```

Comment:
Javadocs could include the @param, and more details, other then that the code is generally very easy to read !